concent_api/core/message_handlers.py

```diff
@@ -153,75 +154,74 @@ def handle_send_ack_report_computed_task(client_message: message.tasks.AckReport
-            raise Http400(
-                "Subtask requestor key does not match current client key. Can't accept your 'AckReportComputedTask'.",
-                error_code=ErrorCode.QUEUE_REQUESTOR_PUBLIC_KEY_MISMATCH,
-            )
+            if subtask.requestor.public_key_bytes != requestor_public_key:
+                raise Http400(
+                    "Subtask requestor key does not match current client key. Can't accept your 'AckReportComputedTask'.",
+                    error_code=ErrorCode.QUEUE_REQUESTOR_PUBLIC_KEY_MISMATCH,
+                )
 
-        if subtask.ack_report_computed_task_id is not None or subtask.reject_report_computed_task_id is not None:
-            raise Http400(
-                "Received AckReportComputedTask but RejectReportComputedTask "
-                "or another AckReportComputedTask for this task has already been submitted.",
-                error_code=ErrorCode.SUBTASK_DUPLICATE_REQUEST,
-            )
-        validate_all_messages_identical([
-            task_to_compute,
-            deserialize_message(subtask.task_to_compute.data.tobytes()),
-        ])
-        new_report_computed_task = None
-        try:
+            if subtask.ack_report_computed_task_id is not None or subtask.reject_report_computed_task_id is not None:
+                raise Http400(
+                    "Received AckReportComputedTask but RejectReportComputedTask "
+                    "or another AckReportComputedTask for this task has already been submitted.",
+                    error_code=ErrorCode.SUBTASK_DUPLICATE_REQUEST,
+                )
             validate_all_messages_identical([
-                report_computed_task,
-                deserialize_message(subtask.report_computed_task.data.tobytes()),
+                task_to_compute,
+                deserialize_message(subtask.task_to_compute.data.tobytes()),
             ])
-        except ConcentValidationError:
-            new_report_computed_task = report_computed_task
-
-        subtask = update_and_return_updated_subtask(
-            subtask                     = subtask,
-            state                       = Subtask.SubtaskState.REPORTED,
-            next_deadline               = None,
-            set_next_deadline           = True,
-            ack_report_computed_task    = client_message,
-            report_computed_task        = new_report_computed_task,
-        )
-        store_pending_message(
-            response_type       = PendingResponse.ResponseType.ForceReportComputedTaskResponse,
-            client_public_key   = provider_public_key,
-            queue               = PendingResponse.Queue.Receive,
-            subtask             = subtask,
-        )
-        logging.log_message_added_to_queue(
-            logger,
-            client_message,
-            requestor_public_key,
-        )
-        return HttpResponse("", status = 202)
+            new_report_computed_task = None
```

Comment:
this part could be extracted to a function (i.e. `substitute_report_computed_task_if_needed`) and get some small comment describing why this is happening

Reply:
Done, hope you will like it